src/main/java/com/manywho/services/dummy/account/repositories/AccountRepository.java

```diff
@@ -18,44 +20,115 @@ public AccountRepository(Sql2o sql2o) {
 
     public Account find(String id) {
         String sql = "SELECT * FROM account WHERE id = :id";
-
         try (Connection connection = sql2o.open()) {
             return connection.createQuery(sql)
                     .addParameter("id", id)
                     .executeAndFetchFirst(Account.class);
         }
     }
 
-    public List<Account> findAll(ListFilter filter) {
-        String sql = "SELECT * FROM account";
+    public List<Account> findAll(ListFilter filter) throws ServiceProblemException {
 
-        try (Connection connection = sql2o.open()) {
+        String sql = "SELECT * FROM account ";
+        if (filter != null && filter.getWhere() != null && filter.getWhere().size() != 0) {
+
+            if (filter.getWhere().size() > 1) {
+                throw new ServiceProblemException(400, "Currently we only support filtering on one value, multiple values have been are provided.");
+            }
+
+            String columnName = filter.getWhere().get(0).getColumnName();
+            String filterByValue = filter.getWhere().get(0).getContentValue();
+            CriteriaType criteriaType = filter.getWhere().get(0).getCriteriaType();
+
+            sql = createSQL(filter, sql, columnName, filterByValue, criteriaType);
+        }
+        try (
+                Connection connection = sql2o.open()) {
             return connection.createQuery(sql)
                     .executeAndFetch(Account.class);
         }
     }
 
+    private String createSQL(ListFilter filter, String sql, String columnName, String value, CriteriaType criteriaType) {
+        if (columnName.equals("id")) {
+            switch (criteriaType) {
+                case GreaterThan:
+                    sql = sql + " WHERE " + columnName + " > " + value;
```

Comment:
the `where` word is added in each case except if there is an exception, you can added before the switch, and do the same in the other switch, also it looks to me a good case to use StringBuilder and pass the StringBuilder instead of a String
It can be used like this:
```       
 StringBuilder stringBuilder = new StringBuilder("SELECT * FROM account ");
        stringBuilder.append(" WHERE ");
        stringBuilder.append("a = b");
        stringBuilder.toString();
```